src/openrct2/entity/Peep.cpp

```diff
@@ -2654,19 +2655,20 @@ int32_t PeepCompare(const EntityId sprite_index_a, const EntityId sprite_index_b
  *
  *  rct2: 0x0069926C
  */
-void PeepUpdateNames(bool realNames)
+void PeepUpdateNames()
 {
     auto& gameState = GetGameState();
-    if (realNames)
-    {
+    auto& config = Config::Get().general;
+
+    if (config.ShowRealNamesOfGuests)
         gameState.Park.Flags |= PARK_FLAGS_SHOW_REAL_GUEST_NAMES;
-        // Peep names are now dynamic
-    }
     else
-    {
         gameState.Park.Flags &= ~PARK_FLAGS_SHOW_REAL_GUEST_NAMES;
-        // Peep names are now dynamic
-    }
+
+    if (config.ShowRealNamesOfStaff)
+        gameState.Park.Flags |= PARK_FLAGS_SHOW_REAL_STAFF_NAMES;
+    else
+        gameState.Park.Flags &= ~PARK_FLAGS_SHOW_REAL_STAFF_NAMES;
 
     auto intent = Intent(INTENT_ACTION_REFRESH_GUEST_LIST);
```

Comment:
Does the intent refresh everything or just the guest as the name implies? Because if it's everything, we can follow-up on  another PR to break down and send smaller intents, but if it's just the guest list, then this is not refreshing what it should, right?

Reply:
```cpp
            case INTENT_ACTION_REFRESH_GUEST_LIST:
                WindowGuestListRefreshList();
                break;
```

It's only forcing the guest list to refresh. I'm not seeing any issues with the staff list, though. Perhaps it isn't cached like the guest list? I've not looked into it much.

Reply:
That seems correct, there is also this comment on StaffList:
https://github.com/OpenRCT2/OpenRCT2/blob/e6e3f5db7696697337b43235d08633beeffe444c/src/openrct2-ui/windows/StaffList.cpp#L214-L216

Maybe we should split the functions just to avoid refreshing the guest list unnecessarily, then?

Reply:
I don't think it's worth it, tbh. This is only called once, when the option is toggled, after all.

Reply:
Your call